tracks/java/exercises/hamming/mentoring.md

```diff
@@ -6,8 +6,9 @@ class Hamming {
 
     Hamming(String leftStrand, String rightStrand) {
         if (leftStrand.length() != rightStrand.length())
-            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
-        this.leftStrand = leftStrand;
+            throw new IllegalArgumentException(leftStrand.length() == 0 ? "left strand must not be empty."
+                    : (rightStrand.length() == 0 ? "right strand must not be empty."
```

Comment:
Using a ternary operator inside of a ternary operator is generally considered bad style / practice in Java.

Since the validation is no longer so trivial, it may be best to have our mentoring notes suggest delegating to another method rather than trying to do everything straight in the constructor. For example, extracting a `validateStrands` method.

```java
private static void validateStrands(String leftStrand, String rightStrand) {
  if (leftStrand.length() == rightStrand.length()) {
    return;
  }
  if (leftStrand.isEmpty()) {
    throw new IllegalArgumentException("left ...");
 }
  if (rightStrand.isEmpty()) {
    throw new IllegalArgumentException("right ...");
 }
  throw new IllegalArgumentException("...");
}
```